BraveWallet/Crypto/Search/SwapTokenSearchView.swift

```diff
@@ -0,0 +1,67 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import SwiftUI
+import BraveCore
+import Shared
+
+private struct TokenView: View {
+  var token: BraveWallet.ERCToken
+  
+  var body: some View {
+    HStack(spacing: 8) {
+      AssetIconView(token: token)
+      VStack(alignment: .leading) {
+        Text(token.name)
+          .fontWeight(.semibold)
+          .foregroundColor(Color(.bravePrimary))
+        Text(token.symbol.uppercased())
+          .foregroundColor(Color(.secondaryBraveLabel))
+      }
+      .font(.footnote)
+    }
+    .padding(.vertical, 8)
+  }
+}
+
+struct SwapTokenSearchView: View {
+  @ObservedObject var swapTokenStore: SwapTokenStore
+
+  @Environment(\.presentationMode) @Binding private var presentationMode
+  
+  enum SwapSearchType {
+    case fromToken
+    case toToken
+  }
+  
+  var searchType: SwapSearchType
```

Comment:
search list for `from` token should not contain the selected `to` token
search list for `to` token should not contain the selected `from` token